Create Plugin: Allow users to adopt CI/CD during update

asimpson opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/plugin-tools/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**What this PR does / why we need it**:

Currently the GitHub workflow files are only generated during the `generate` command. Running `update` doesn't do anything with workflows and for good reason since users who run `update` _probably_ have CI/CD configured already. However, I ran into a situation recently where a datasource would have benefited from being able to "adopt" the workflow files.

#541 and #543 will potentially make this less destructive and we could explore just printing the necessary `yml` config to `stdout` at that point since the line count of the config will be much smaller.

With this change users can now run `npx @grafana/create-plugin@latest cicd` to explicitly adopt the Github Actions workflows.

academo reviewed (commented):
Code-wise this is correct but I'd like to hear input from @sympatheticmoose on this one. 

There's a risk that people would just press "Y" because we already have so many questions

Note that people will have a git project and reverting these changes might be trivial in case of mistake, so it is not a big problem to solve if you pressed the wrong key.

asimpson commented:
@academo @davemoore is correct, we settled on something like: `npx @grafana/create-plugin@latest cicd`. I can update this PR today hopefully with this change. 

I do think we could be more aggressive in the `update` command because of git like you said, but also because if someone is running `update` there's a built-in assumption (to me) that things will change :man_shrugging:

asimpson committed:
move cicd to separate subcommand

asimpson committed:
fix linter and issues with previous commits

asimpson committed:
update docs

asimpson commented:
@sunker so sorry for the delay in getting back to this. I added docs and refactored things.

The new question since I created the original draft of this PR is should we be doing it this way given [the new action](https://github.com/grafana/plugin-actions/blob/main/build-plugin/README.md)? Will #742 impact how this works?

sunker reviewed (commented):
Sorry @asimpson I missed this. Changes looks good to me, but would like someone else from the team to have a look at it too.

sympatheticmoose commented:
@asimpson thanks again for this - theres some upcoming changes with generation which will cause issues with this - leave with us for now, but please reach out if this is blocking you

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/plugin-tools?pullRequest=569) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/plugin-tools?pullRequest=569) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/plugin-tools?pullRequest=569) it.</sub>
